### ISSUE TITLE
Vapid tests intermittently fail with "Error: Vapid private key should be 32 bytes long when decoded."

### ISSUE THREAD

**Issue by marco-c** [web-push-libs/web-push#295]
We have seen this in multiple instances (e.g. in #286 and in #294). I'm attaching some logs before they expire.
[log1.txt](https://github.com/web-push-libs/web-push/files/1307961/202306396.txt)
[log2.txt](https://github.com/web-push-libs/web-push/files/1307966/log2.txt)
[log3.txt](https://github.com/web-push-libs/web-push/files/1307967/log3.txt)

**Comment by iEchoic** [comment]
I am also having this issue:

`Vapid private key should be 32 bytes long when decoded`

**Comment by gauntface** [comment]
Stupid way to trigger this:

```
for (let i = 0; i < 400; i += 1) {
    test(`generate vapid keys ${i}`, function() {
      const keys = webPush.generateVAPIDKeys();
      assert(keys.privateKey);
      assert(keys.publicKey);

      assert.equal(typeof keys.privateKey, 'string');
      assert.equal(typeof keys.publicKey, 'string');

      assert.equal(urlBase64.decode(keys.privateKey).length, 32);
      assert.equal(urlBase64.decode(keys.publicKey).length, 65);
    });
  }
```

**Comment by gauntface** [comment]
The issue is intermittent because the failing vapid keys need to be padded:

Printing out the strings via `curve.getPrivateKey('base64')` prints out examples where the failing keys have double `=` on the end (i.e. padding bits).

```
Valid:    LHv45HHq4Ly3+W+w0LW8g0iUp/fVw58Nylk0cxfFsuk= 44
Invalid:  sHY2lcuXBT3TR7m9JZvmGv8v8Zg5hEZHiqX4AnBvMA== 44
```
My guess is that we should be adding a padding bit on buffer or the urlsafe-base64 encoded string.

**Comment by marco-c** [comment]
I've published version 3.2.4 with the fix by @gauntface.

**Comment by shakti2505** [comment]
Vapid private key should be 32 bytes long when decoded.
I am facing this error when sending notification from server I have removed the space in the keys and using web-push 3.6.7
Please guide me, Thanks
